test/unit/range/view/view_single_pass_input_test.cpp

```diff
@@ -0,0 +1,378 @@
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <gtest/gtest.h>
+
+#include <seqan3/range/view/single_pass_input.hpp>
+#include <seqan3/range/concept.hpp>
+#include <seqan3/range/view/concept.hpp>
+
+#include <vector>
+#include <type_traits>
+
+#include <range/v3/view/zip.hpp>
+#include <range/v3/view/take.hpp>
+#include <range/v3/istream_range.hpp>
+
+template <typename T>
+class single_pass_input : public ::testing::Test
+{};
+
+// add all <out_rng,in_rng> pairs here.
+using underlying_range_types = ::testing::Types<std::vector<char>,
+                                                std::vector<int>,
+                                                std::vector<char> const,
+                                                ranges::istream_range<char>,
+                                                ranges::istream_range<int>>;
+
+TYPED_TEST_CASE(single_pass_input, underlying_range_types);
+
+using namespace seqan3;
+
+template <typename rng_type>
+auto get_data()
+{
+    if constexpr (std::is_same_v<std::remove_cv_t<rng_type>, std::vector<char>>)
+    {
+        return rng_type{'1','2','3','4','5'};
+    }
+    else if constexpr (std::is_same_v<std::remove_cv_t<rng_type>, std::vector<int>>)
+    {
+        return rng_type{1, 2, 3, 4, 5};
+    }
+    else if constexpr (std::is_same_v<std::remove_cv_t<rng_type>, ranges::v3::istream_range<char>>)
+    {
+        return std::istringstream{"12345"};
+    }
+    else /*(std::is_same_v<rng_type, ranges::v3::istream_range<int>>)*/
+    {
+        return std::istringstream{"1 2 3 4 5"};
+    }
+}
```

Comment:
I think this stuff would be better off in a fixture?

Reply:
Good point.